### PR TITLE
HTCONDOR-2475 Remove StringLists from gangliad

### PR DESCRIPTION
**Pull request by GregThain**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/gangliad/statsd.cpp [hunk 1]
```diff
@@ -963,20 +976,19 @@ StatsD::publishMetrics( int /* timerID */ )
 	}
 
 	// Set query constraint to only fetch ad types needed
-	if( !m_target_types.contains_anycase("any") ) {
+	if (!contains_anycase(m_target_types,"any")) {
 		if( !m_requirements.empty() ) {
 			query.addANDConstraint(m_requirements.c_str());
 		}
 	}
 	else {
```
Comment (GregThain) [inline src/gangliad/statsd.cpp:984]: Please pay particular attention to the follow block of code, as I believe there was a bug on the trunk -- a missing call to `rewind`.  The new code assumes there was a rewind there.

Comment (JaimeFrey) [reply src/gangliad/statsd.cpp:984]: Agreed that it looks like a `rewind()` is missing. And I think the `contains_anycase()` conditional is backwards. Check out commit 30663964e20.

Comment (GregThain) [reply src/gangliad/statsd.cpp:984]: Huh.  Look at that.  We need to push for a good regression test for the gangliad, maybe even if it just writes ads it pulls out of the collector to a file.

Comment (GregThain) [reply src/gangliad/statsd.cpp:984]: Almost feel like the condition should be something like `m_target_types.size() == 1 && m_target_types[0] == "any"`.

Seems odd to have "any" mixed in with other types.

Comment (JaimeFrey) [reply src/gangliad/statsd.cpp:984]: That seems like a mistake in the config. But I wouldn't ignore the 'any' because there are other types in the list.

**Review by JaimeFrey — APPROVED** [review]
LGTM

**Pull request merged**